regolith/helpers/l_abstracthelper.py

```diff
@@ -74,25 +82,37 @@ def sout(self):
         rc = self.rc
         presentations = self.gtx["presentations"]
 
+        if (not rc.author) and (not rc.year) and (not rc.loc_inst) and (not rc.title):
+            return
+
         for presentation in presentations:
             if rc.title is not None:
                 if rc.title.casefold() not in presentation.get('title').casefold():
                     continue
-            if rc.location is not None and 'location' in presentation:
-                if rc.location.casefold() not in presentation.get('location').casefold():
-                    continue
-            if rc.location is not None and 'location' not in presentation:
-                continue
+            if presentation['type'] == 'seminar' or presentation['type'] == 'colloquium':
+                if rc.loc_inst is not None:
+                    if rc.loc_inst.casefold() not in presentation.get('institution').casefold():
+                        continue
+            else:
+                if rc.loc_inst is not None and 'location' in presentation:
```

Comment:
ugly code here too.  If I were you, I would take a step back and try and think what is a more elegant way to do this whole activity of filtering for the location and start over.  I am not sure the logic, but when I see tortuous code like this I am almost sure there is a better way to do it.